apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -328,41 +358,25 @@ const FormController = props => {
     setGlobalError(false);
     setSubmitting(true);
 
-    // [MEG] TODO: only do a post if it's a new application
-    $.ajax({
-      method: 'POST',
-      url: apiEndpoint,
-      contentType: 'application/json',
-      dataType: 'json',
-      data: JSON.stringify(serializeFormData())
-    })
-      .done(data => {
-        sessionStorage.removeItem(sessionStorageKey);
-        onSuccessfulSubmit(data);
-      })
-      .fail(data => {
-        if (
-          data.responseJSON &&
-          data.responseJSON.errors &&
-          data.responseJSON.errors.form_data
-        ) {
-          if (data.responseJSON.general_error) {
-            setErrors(data.responseJSON.errors.form_data);
-            setErrorHeader(data.responseJSON.general_error);
-            setGlobalError(true);
-          } else {
-            // if the failure was a result of an invalid form, highlight the errors
-            // and display the generic error header
-            setErrors(data.responseJSON.errors.form_data);
-            setErrorHeader(i18n.formErrorsBelow());
-          }
-        } else {
-          // Otherwise, something unknown went wrong on the server
-          setGlobalError(true);
-          setErrorHeader(i18n.formServerError());
-        }
-        setSubmitting(false);
-      });
+    const handleSuccessfulSubmit = data => {
+      sessionStorage.removeItem(sessionStorageKey);
+      onSuccessfulSubmit(data);
+    };
+
+    const handleRequestFailure = data => {
+      if (get(data, 'responseJSON.errors.form_data')) {
+        setGlobalError(true);
+        setErrors(data.responseJSON.errors.form_data);
+        setErrorHeader(i18n.formErrorsBelow());
+      } else {
+        // Otherwise, something unknown went wrong on the server
+        setGlobalError(true);
+        setErrorHeader(i18n.formServerError());
+      }
+      setSubmitting(false);
+    };
+
+    makeRequest(handleSuccessfulSubmit, handleRequestFailure);
```

Comment:
if I am reading correctly, this would become 
```suggestion
    makeRequest.done(handleSuccessfulSubmit).fail(handleRequestFailure);
```
apologies if I missed a nuance here, though!